subprojects/plugin-use/src/integTest/groovy/org/gradle/plugin/use/AlreadyOnClasspathPluginUseIntegrationTest.groovy

```diff
@@ -0,0 +1,323 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.plugin.use
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.BuildOperationsFixture
+
+class AlreadyOnClasspathPluginUseIntegrationTest extends AbstractIntegrationSpec {
+
+    def operations = new BuildOperationsFixture(executer, temporaryFolder)
+
+    def setup() {
+        settingsFile << "rootProject.name = 'root'\n"
+    }
+
+    def "can request buildSrc plugin"() {
+
+        given:
+        withBinaryPluginBuild("buildSrc")
+
+        and:
+        settingsFile << "include('a')\n"
+
+        and:
+        buildFile << requestPlugin("my-plugin")
+        file("a/build.gradle") << requestPlugin("my-plugin")
+
+        when:
+        succeeds "help"
+
+        then:
+        outputContains(appliedPluginOutput())
+
+        and:
+        operations.hasOperation("Apply plugin my-plugin to root project 'root'")
+        operations.hasOperation("Apply plugin my-plugin to project ':a'")
+    }
+
+    def "can request non-core plugin already applied to parent project"() {
+
+        given:
+        settingsFile << "include('a')\n"
+
+        and:
+        buildFile << requestPlugin("com.gradle.plugin-publish", "0.9.8")
```

Comment:
⭕️  My preference would be to use a plugin local to the test, like publishing a small plugin to a local ivy repository during `setup` or something.